src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs

```diff
@@ -86,101 +85,59 @@ public bool ExecuteCommand(ValueTrackingEditorCommandArgs args, CommandExecution
                     {
-                        // TODO: Show error dialog
                         return;
                     }
 
-                    var syntaxTree = document.GetRequiredSyntaxTreeSynchronously(cancellationToken);
-                    var location = Location.Create(syntaxTree, textSpan);
-
-                    await ShowToolWindowAsync(args.TextView, selectedSymbol, location, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+                    await ShowToolWindowAsync(args.TextView, document, items, cancellationToken).ConfigureAwait(false);
                 });
 
             return true;
         }
 
-        private static async Task<ISymbol?> GetSelectedSymbolAsync(TextSpan textSpan, Document document, CancellationToken cancellationToken)
-        {
-            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var selectedNode = root.FindNode(textSpan);
-            if (selectedNode is null)
-            {
-                return null;
-            }
-
-            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var selectedSymbol =
-                semanticModel.GetSymbolInfo(selectedNode, cancellationToken).Symbol
-                ?? semanticModel.GetDeclaredSymbol(selectedNode, cancellationToken);
-
-            if (selectedSymbol is null)
-            {
-                return null;
-            }
-
-            return selectedSymbol switch
-            {
-                ILocalSymbol
-                or IPropertySymbol { SetMethod: not null }
-                or IFieldSymbol { IsReadOnly: false }
-                or IEventSymbol
-                or IParameterSymbol
-                => selectedSymbol,
-
-                _ => null
-            };
-        }
-
-        private async Task ShowToolWindowAsync(ITextView textView, ISymbol selectedSymbol, Location location, Solution solution, CancellationToken cancellationToken)
+        private async Task ShowToolWindowAsync(ITextView textView, CodeAnalysis.Document document, ImmutableArray<ValueTrackedItem> items, CancellationToken cancellationToken)
         {
-            var item = await ValueTrackedItem.TryCreateAsync(solution, location, selectedSymbol, cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (item is null)
-            {
-                return;
-            }
-
             var toolWindow = await GetOrCreateToolWindowAsync(textView, cancellationToken).ConfigureAwait(false);
             if (toolWindow?.ViewModel is null)
             {
                 return;
             }
 
-            var valueTrackingService = solution.Workspace.Services.GetRequiredService<IValueTrackingService>();
+            var rootItemMap = items.GroupBy(i => i.Parent, resultSelector: (key, items) => (parent: key, children: items.ToImmutableArray())).ToImmutableArray();
```

Comment:
I'd avoid ToImmutableArray here (both) since this grouping is only used locally.